examples/hosts/app-integration/external-controller/src/app.ts

```diff
@@ -64,14 +63,42 @@ const containerSchema = {
     },
 };
 
-async function initializeNewContainer(container: IFluidContainer): Promise<void> {
+function createDiceRollerControllerProps(map: SharedMap): DiceRollerControllerProps {
+    return {
+        get: (key: string) => map.get(key) as number,
+        set: (key: string, value: any) => () => {map.set(key, value);},
+        on(event: "valueChanged", listener: (args: IValueChanged) => void) {
+            map.on(event, listener);
+            return this;
+        },
+        off(event: "valueChanged", listener: (args: IValueChanged) => void) {
+            map.on(event, listener);
+            return this;
+        },
+    };
+}
+
+function createDiceRollerControllerPropsFromContainer(container: IFluidContainer):
+[DiceRollerControllerProps, DiceRollerControllerProps] {
+    const diceRollerController1Props: DiceRollerControllerProps = createDiceRollerControllerProps(
+        container.initialObjects.map1);
+    const diceRollerController2Props: DiceRollerControllerProps = createDiceRollerControllerProps(
+        container.initialObjects.map2);
```

Comment:
You may need to cast these to `SharedMap` if they're just `IFluidLoadable`.  I don't really think this was supposed to be required, but it seems below we were doing it.

Reply:
done, thanks for the help